expat/CMakeLists.txt

```diff
@@ -58,6 +58,14 @@ if(NOT WIN32)
         set(XML_DEV_URANDOM 0)
     endif(XML_DEV_URANDOM)
 endif()
+if (NOT WIN32)
+    option(HAVE_GETRANDOM "Define to bypass auto-detection of getrandom function from include <sys/random.h> ." OFF)
+    if(HAVE_GETRANDOM)
+        set(HAVE_GETRANDOM 1)
+    else(HAVE_GETRANDOM)
+        set(HAVE_GETRANDOM 0)

```

Comment:
This will delete auto-detection always, not just for `-DHAVE_GETRANDOM`.

What I expect to work without breaking autodetection (and being run after auto-detection), is this:
```
option(WITH_GETRANDOM "helpful text here" ON)
if(NOT WITH_GETRANDOM)
    set(HAVE_GETRANDOM 0)
endif()
```

Reply:
May be i didnt understand the requirement correctly. Could you please help me understand this.

I was assuming that there should be a option to pass to CMake or AutoTools to disable/enable getrandom, hence all the changes are made.

From your comments the confusing part is that the below code has to be enabled, it means that auto detection should be there.
check_symbol_exists("getrandom" "sys/random.h" HAVE_GETRANDOM)

Reply:
As I understand (please correct me if i wrong), there should be a option to disable even after auto detection of getrandom.

Reply:
I have committed another change with the same pull request for CMake: Please review it

Reply:
I was hoping for three modes: enforce use, prohibit use, auto-detect (default) in a single variable like Autotools can do. CMake does not seem to support "tribools" like this out of the box.
I have built upon your commit now to make it work nevertheless and merged all related commits to `master` — merge commit bf926a6edc15abdb8d113b72580e68483d6b47cd — to save some ping-pong. If you find shortcomings with the tribool approach taken I'm happy to discuss more in a new ticket. I hope that works for you.